### ISSUE TITLE
IntelliSense problems

### ISSUE THREAD

**Issue by svipben** [flowtype/flow-for-vscode#97]
Hello,

This plugin is really awesome, but I noticed a problem, it was since I first discovered this plugin, so maybe more than 3 months. It happens for all variables, classes and so on. The problem is you get duplicate autocomplete suggestions which is mixed with VS Code and Flow, my suggestion is: if Flow finds autocompletion it disables VS Code to don't duplicate it, because right now it's really messy... Also sometimes if no VS Code suggestions finded, Flow suggestions will be almost at the bottom, there's no normal arrangement for Flow suggestions.

As you can see duplicated suggestions and this happens a lot, not only for variables...
![screenshot from 2017-03-25 14-57-57](https://cloud.githubusercontent.com/assets/10441177/24322379/e4fe692a-116b-11e7-9757-23f66fab7a39.png)

The most annoying part, sometimes you think there's no suggestions, but then you scroll down you will find them which isn't good, because they are mixed with just word suggestions which are found in file, there's no normal arrangement for Flow suggestions...
![screenshot from 2017-03-25 15-04-05](https://cloud.githubusercontent.com/assets/10441177/24322415/9b5d94b6-116c-11e7-96b8-37f8d841507e.png)

In short: better always use Flow suggestions (if there's no Flow suggestions for that variable, etc. use VS Code then) and do normal arrangement of Flow suggestions.

Thanks,
svipben

**Comment by vincaslt** [comment]
I agree with this, I noticed double suggestions multiple times, and it's really starting to clutter the dropdown.

**Comment by orta** [comment]
It's because of VS Code's TypeScript interpreter which cannot be turned off alas, closing as dupe of https://github.com/flowtype/flow-for-vscode/issues/50

**Comment by svipben** [comment]
@orta So both problems are impossible to fix? (duplicate suggestions and bad suggestions arrangement)

**Comment by orta** [comment]
Yep, from this extension side

**Comment by svipben** [comment]
@orta What does VS Code team says? Maybe I need to create issue about that?

**Comment by orta** [comment]
There are linked issues in #50 - would recommend looking through those 👍

**Comment by yeze322** [comment]
After reading [#17068](https://github.com/Microsoft/vscode/issues/17068), I found it's not possible to solve this issue in a short time. What a pity :(

I'm wondering if Atom has the same problem?

**Comment by yeze322** [comment]
Is it possible to add prefix like `$` or `_` automatically before property names, so they can appear before other suggestions?

If so, there may has a tricky solution:

- Firstly, the flow plugin implicitly inserts `$` before each properties. *(e.g. `id` -> `$id`)*
- Once an suggestion is selected, remove the `$` prefix. *(e.g. `$id` -> `id`)*
- Fianlly, submit the string without prefix. We still get `id` inserted.

**Just had a try:**
- The origin code

![image](https://cloud.githubusercontent.com/assets/8528761/25070379/f1d4b608-22ce-11e7-8dce-26702bd09cda.png)

- `$` works well:

![image](https://cloud.githubusercontent.com/assets/8528761/25070214/6b7d2624-22cb-11e7-8be8-5568af19b6d2.png)

- But `_` still has bug

![image](https://cloud.githubusercontent.com/assets/8528761/25070224/96730efc-22cb-11e7-8ba5-b7b3132c214d.png)

**Comment by svipben** [comment]
Opened issue in VS Code here: https://github.com/Microsoft/vscode/issues/24974.

**Comment by davnicwil** [comment]
For anyone landing here now, vscode now supports this.

In user settings 

```
"editor.wordBasedSuggestions": false
```

All flow suggestions now appear on top. The duplicates issue is still there and I haven't found how to fix that, but still the situation now is much better.

![image](https://user-images.githubusercontent.com/9915841/38821305-0c145620-4198-11e8-886e-a88c9b273ff3.png)

**Comment by svipben** [comment]
@davnicwil you can now disable built-in extensions like TypeScript/JavaScript, etc.

https://code.visualstudio.com/updates/v1_21#_extensions

**Comment by davnicwil** [comment]
@svipben thanks for this, this works nicely -- however when you turn off the built in TypeScript/JavaScript extension you lose the ability to cmd+click imports and link to the files, which is not ideal. I guess it is still a trade off for now between two annoyances.

**Comment by svipben** [comment]
To be honest I believe it needs a total rewrite of this extension or completely to use Atom if you're using Flow. If you're using VSC you can use vanilla JS and/or TypeScript. But with Flow, you will not get such benefits as in Atom because of support, stability, and features. I know this sounds odd, but that's true...

I tried to use Flow at 2017 March and the behavior how bad it is for Visual Studio Code exists till now at 2018 May. 1 year and nothing changed :) Issues are growing because Flow team doesn't care about Visual Studio Code.

TL;DR: If you're using Flow switch to Atom or WebStorm. If you're using TypeScript switch to Visual Studio Code or WebStorm.

**Comment by thymikee** [comment]
I'd rather like to make VSC integration better. There are a lot of moving parts, but we'll get there 🤞.

**Comment by wi-ski** [comment]
This sucks really bad?

**Comment by davnicwil** [comment]
@svipben, and everyone else watching this, I've implemented a config to disable the typescript autocomplete suggestions. It's solving this problem for me locally and I've [submitted a PR]((https://github.com/Microsoft/vscode/pull/58011)).

If you want this feature, please comment there to show your support!

**Comment by orta** [comment]
@davnicwil this is a smart way to get this fixed 👍

**Comment by davnicwil** [comment]
This landed by the way, in 1.28.0

Set  `typescript.suggest.enabled: false` to turn off TypeScript autocomplete suggestions completely.

Release notes here https://code.visualstudio.com/updates/v1_28#_new-settings-for-jsts-suggestions

**Comment by orta** [comment]
Can any extension resolve options like this at runtime? We could recommend turning it off in project settings if it's not set in an update.

**Comment by goxr3plus** [comment]
Why IntelliSense shows JavaScript methods on reccomendation for Java Code ? How i can disable this ...
